extensions/amp-access/0.1/amp-access.js

```diff
@@ -351,21 +273,16 @@ export class AccessService {
 
   /** @private */
   startInternal_() {
-    dev().fine(TAG, 'config:', this.type_, this.loginConfig_,
-        this.adapter_.getConfig());
-
     // TODO(dvoytenko, #3742): This will refer to the ampdoc once AccessService
     // is migrated to ampdoc as well.
     Services.actionServiceForDoc(this.ampdoc).installActionHandler(
         this.accessElement_, this.handleAction_.bind(this));
 
-    // Calculate login URLs right away.
-    this.buildLoginUrls_();
-
-    // Start sign-in.
-    this.signIn_.start();
+    for (let i = 0; i < this.sources_.length; i++) {
+      this.sources_[i].start();
+    }
 
-    // Start authorization XHR immediately.
+    // Run authorization immediately
```

Comment:
Hmm. So the cycle above starts sources which in turn run their respective authorizations. And this one runs authorization as well. So they run twice?

Reply:
fixed, i believe. `start()` in `AccessSource` no longer calls `runAuthorization` and `start()` in `AmpAccess` blocks on `whenFirstVisible`